src/HttpClientFactory/Polly/src/DependencyInjection/PollyServiceCollectionExtensions.cs

```diff
@@ -30,17 +32,18 @@ public static IPolicyRegistry<string> AddPolicyRegistry(this IServiceCollection
         // single instance registered using all the interfaces.
         var registry = new PolicyRegistry();
 
-        services.AddSingleton<IConcurrentPolicyRegistry<string>>(registry);
-        services.AddSingleton<IPolicyRegistry<string>>(registry);
-        services.AddSingleton<IReadOnlyPolicyRegistry<string>>(registry);
+        services.TryAddEnumerable(ServiceDescriptor.Singleton<IPolicyRegistry<string>>(registry));
```

Comment:
Out of curiosity, why change the order?

Reply:
> Out of curiosity, why change the order?

There are 3 extension methods that always register the same 3 types and I thought it would be a bit better (more consistent and intention clear) to have the same registration order in all methods: ```IPolicyRegistry``` then ```IReadOnlyPolicyRegistry``` then ```IConcurrentPolicyRegistry```. I saw this order already being used in the 2nd extension method and it made the most sense to me.
I tried not to be careless and I thought if changing the order would break something but I couldn't find an issue.

Reply:
Makes sense to me.  Thanks for the explanation!